collector/lib/CollectorConfig.cpp

```diff
@@ -438,17 +453,88 @@ void CollectorConfig::HandleConfig(const std::filesystem::path& filePath) {
     YAML::Node yamlConfig = YAML::LoadFile(filePath);
     YamlConfigToConfig(yamlConfig);
   } catch (const YAML::BadFile& e) {
-    CLOG(FATAL) << "Failed to open the configuration file: " << filePath
-                << ". Error: " << e.what();
+    std::string errorMsg = "Failed to open the configuration file: " + filePath.string() + ". Error: " + e.what();
+    LogRuntimeConfigError(errorMsg);
```

Comment:
Instead of this, you can have `HandleConfig` return a boolean for file parsing success.
```cpp
CLOG(ERROR) << "Failed to open the configuration file: " << filePath
                << ". Error: " << e.what();
return false;
```

Then you can use this returned value during initialization to do a `CLOG(FATAL)` and ignore it during regular operation. This also allows dropping the `LogRuntimeConfigError` method, which is not very ergonomic IMO.